platform/entanglement/test/services/LinkServiceSpec.js

```diff
@@ -197,11 +215,22 @@ define(
                     whenComplete = jasmine.createSpy('whenComplete');
                     returnPromise.then(whenComplete);
 
-                    mutationPromise.resolve();
+                    addPromise.resolve(linkedObject);
                     persistencePromise.resolve();
                     compositionPromise.resolve([linkedObject]);
                     expect(whenComplete).toHaveBeenCalledWith(linkedObject);
                 });
+
+                it("throws an expection when performed on invalid inputs", function () {
+                    function perform() {
+                        linkService.perform(object, parentObject);
+                    }
+
+                    mockPolicyService.allow.andReturn(true);
```

Comment:
Similar to CopyServiceSpec, spying on the validate method instead of the policy service would tighten the focus of this unit test.